managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -1619,7 +1615,9 @@ public void asyncClose(final AsyncCallbacks.CloseCallback callback, final Object
         // hence we write it as -1. The cursor ledger is deleted once the z-node write is confirmed.
         ManagedCursorInfo info = ManagedCursorInfo.newBuilder().setCursorsLedgerId(-1)
                 .setMarkDeleteLedgerId(markDeletePosition.getLedgerId())
-                .setMarkDeleteEntryId(markDeletePosition.getEntryId()).build();
+                .setMarkDeleteEntryId(markDeletePosition.getEntryId())
+                .addAllIndividualDeletedMessages(buildIndividualDeletedMessageRanges())
```

Comment:
Could we skip this line if we have no individually deleted messages? We should avoid to create an empty list in that case.